src/Management/test/Endpoint.Test/Actuators/CloudFoundry/PermissionsProviderTest.cs

```diff
@@ -41,8 +43,64 @@ public async Task GetPermissionsTest()
         };
 
         response.Content = JsonContent.Create(permissions);
-        EndpointPermissions result = await permissionsProvider.GetPermissionsAsync(response, TestContext.Current.CancellationToken);
-        Assert.Equal(EndpointPermissions.Full, result);
+        EndpointPermissions result = await permissionsProvider.ParsePermissionsAsync(response, TestContext.Current.CancellationToken);
+        result.Should().Be(EndpointPermissions.Full);
+    }
+
+    [InlineData("unavailable", HttpStatusCode.ServiceUnavailable, PermissionsProvider.CloudfoundryNotReachableMessage)]
+    [InlineData("not-found", HttpStatusCode.Unauthorized, PermissionsProvider.InvalidTokenMessage)]
+    [InlineData("unauthorized", HttpStatusCode.Unauthorized, PermissionsProvider.InvalidTokenMessage)]
+    [InlineData("forbidden", HttpStatusCode.Forbidden, PermissionsProvider.AccessDeniedMessage)]
+    [InlineData("timeout", HttpStatusCode.ServiceUnavailable, PermissionsProvider.CloudfoundryNotReachableMessage)]
+    [InlineData("exception", HttpStatusCode.InternalServerError, "Exception of type 'System.Net.Http.HttpRequestException' was thrown.")]
+    [InlineData("no_sensitive_data", HttpStatusCode.OK, "")]
+    [InlineData("success", HttpStatusCode.OK, "")]
+    [Theory]
+    public async Task CloudControllerScenarioTesting(string cloudControllerResponse, HttpStatusCode steeltoeStatusCode, string? expectedMessage)
```

Comment:
Same comments about the test name and `cloudControllerResponse` parameter. Additionally, `expectedMessage` should not be nullable -or- tests should pass null instead of "".

Reply:
Does this one look ok now?

Reply:
Should align the name with the other conversation.